linked_list/implementation.py

```diff
@@ -1,40 +1,235 @@
+from .node import Node
 from .interface import AbstractLinkedList
-
+from copy import deepcopy
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        
+        if elements == None:
+            self.start = self.end
+        elif len(elements) == 1:
+            # Build a single node here
+            if isinstance(elements, list):
+                newnode = Node(elements[0])
+                self.start = newnode
+                self.end = newnode
+            else:
+                newnode = Node(elements)
+                self.start = newnode
+                self.end = newnode
+        elif len(elements) > 1:
+            # Test For loop to initialize nodes
+            self.start = Node(elements[0])
+            current_node = self.start
+            for item in elements[1:-1]:
+                next_node = Node(item)
+                current_node.next = next_node
+                current_node = next_node
+            self.end = Node(elements[-1])
+            current_node.next = self.end
+        self.cur_pointer = self.start
+                
     def __str__(self):
-        pass
+        result = []
+        if self.start == None:
+            return "[]"
+        cur = self.start
+        while cur != None:
+            result.append(cur.elem)
+            cur = cur.next
+        return str([(i) for i in result])
```

Comment:
Not really sure why you're making a list comprehension here and wrapping the items in results in parenthesis to make them tuples? I'll try to run this later and see whats up (or if you can tell me whats going on briefly would be great ;))